src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -709,6 +659,7 @@ manyTill (Fold fstep finitial fextract)
             Error _ -> do
                 rR <- initialL
                 return $ Continue (cnt + 1) (ManyTillL 0 fs rR)
+    -- XXX the cnt is being used only by the assert
```

Comment:
You can remove `cnt`. I think this remained when I removed out `Partial1` and `Done1`.

Reply:
Let's keep the assert for now, its good for debugging.